src/wildcard.jl

```diff
@@ -0,0 +1,29 @@
+struct WildcardDimensions{R} <: AbstractDimensions{R}
+    # We simply choose Bool since it will get promoted
+    WildcardDimensions() = new{Bool}()
+end
+
+constructorof(::Type{<:WildcardDimensions}) = WildcardDimensions
+dimension_names(::Type{<:WildcardDimensions}) = ()
+
+# Wildcard dimensions are always compatible:
+Base.:(==)(::AbstractDimensions, ::WildcardDimensions) = true
+Base.:(==)(::WildcardDimensions, ::AbstractDimensions) = true
+
+# For any zero/oneunit specified on the type, we return a wildcard dimension instead:
+Base.zero(::Type{Q}) where {T,Q<:UnionAbstractQuantity{T}} = constructorof(Q)(zero(T), WildcardDimensions())
+
+# For propagation rules, we assume all relevant dimensions are zero
+Base.getproperty(::WildcardDimensions, ::Symbol) = zero(Bool)
+Base.iszero(::WildcardDimensions) = true
```

Comment:
These lines seem too dangerous as discussed. Maybe `getproperty` should just be replaced with an error: (this is not very carefully considered, just a thought)
```suggestion
Base.getproperty(::WildcardDimensions, ::Symbol) = error("Cannot get dimensions of quantity with WildcardDimensions. This is probably because the quantity was created via `zero(T)` for a type `T`. Instead, use `zero(x)` where `x` is a quantity with the desired dimensions.")
```